components/admission-webhook/pkg/apis/settings/v1alpha1/poddefault_types.go

```diff
@@ -53,6 +53,9 @@ type PodDefaultSpec struct {
 	// VolumeMounts defines the collection of VolumeMount to inject into containers.
 	// +optional
 	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
+
+	// ObjectMeta defines the metadata to inject labels and annotations from into the metadata field of a pod.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Did you consider having separate fields for labels & annotations rather than a metadata field?
metadata has a bunch of fields
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#objectmeta-v1-meta

If we include ObjectMeta does that create confusion about what happens for other fields (e.g. name, namespace, ownerReferences, finalizers etc...)?

Reply:
How about introducing the whole field but only use those certain parts?
eg StatefulSet uses PersistentVolumeClaim as the type for volumeClaimTemplates, even if it doesn't use all the fields.

We could also maybe use some of them in the future?

Reply:
The downside of using ObjectMeta is that the API is no longer self-specifying and well defined. A user has to read the docs in order to understand the behavior; i.e. which fields are copied and which are ignored. Whereas if we have fields labels and annotations that are copied then it is arguably more self explanatory.

Furthermore we are in effect saying that the behavior for other fields e.g "finalizers, name, namespace, ownerReferences" are undefined. Its not clear whether these fields are intentionally excluded or they are excluded only because we did a partial implementation in the need of expediency.

Do we have a strong reason to expect that we will want to include these fields in the future? finalizers seems like the only other field that we might need to set in the future. Even then I'm not sure if setting finalizers on pods as opposed to the owning object is common.

For all the other fields (name, uid, ownerReferences) etc... setting them in podDefaults seems incorrect.

Reply:
Yeah I wasn't sure either about using ObjectMeta but with only the annotations/labels fields (what I'm doing here and what @yanniszark describes unless I'm mistaken), or add annotations/labels as separate fields. I've went with using ObjectMeta because having a struct for metadata as oppose to just two map[string]string seemed nicer, but not feeling strongly that way.
I'll update the PR to use Annotations and Labels fields instead.

Reply:
Thanks. I think this is better.